src/components/calcite-action-menu/calcite-action-menu.tsx

```diff
@@ -0,0 +1,338 @@
+import { Component, Host, h, Element, Prop, Watch, State } from "@stencil/core";
+import { CSS, ICONS, TEXT } from "./resources";
+import { focusElement } from "../../utils/dom";
+import { VNode } from "@stencil/core/internal";
+import { getRoundRobinIndex } from "../../utils/array";
+import { PopperPlacement } from "../../utils/popper";
+import { Placement } from "@popperjs/core";
+import { guid } from "../../utils/guid";
+
+const SUPPORTED_BUTTON_NAV_KEYS = ["ArrowUp", "ArrowDown"];
+const SUPPORTED_MENU_NAV_KEYS = ["ArrowUp", "ArrowDown", "End", "Home"];
+const MENU_ANIMATION_DELAY_MS = 50;
+
+@Component({
+  tag: "calcite-action-menu",
+  styleUrl: "calcite-action-menu.scss",
+  shadow: true
+})
+/**
```

Comment:
Nitpick: can you move the JSDoc above `@Component` to keep the decorator next to its target?